cli/remote_build.go

```diff
@@ -946,7 +1007,15 @@ func (cmd *RemoteCmd) submitBuild(project *Project, tagMap map[string]string) er
 					if !buildSuccess {
 						buildSuccess = strings.Count(string(msg), "-- BUILD SUCCEEDED --") > 0
 					}
+					if !buildFailed {
+						buildFailed = strings.Count(string(msg), "-- BUILD FAILED --") > 0
+						if !buildFailed {
+							buildFailed = strings.Count(string(msg), "Patch '' didn't apply cleanly") > 0
```

Comment:
This is okay at the moment but is a little fragile, is there a better way for us to communicate an error? Maybe the dispatcher should have a status channel too in addition to a log stream channel

Reply:
I was talking about that with @amyhua sometime ago. Dispatcher and/or the API should send a structured/typed message communicating the states transitions. @nictuku made a comment about it too.